### PR TITLE
Harvest Temple: added red/spread bait mechanics, display orb push hit count, added giants phase

### PR DESCRIPTION
**Pull request by asterius1**
Added few things to Harvest Temple replay

## Red Bait, Spread Bait mechanics

This one helps to quickly check which people forget to bait their spreads and with added bonus of showing when spread and red appear on graph.

## Orb hit count

I override player dmg numbers on push orbs (purification 1, 2, 3). All strike damage from player is what pushes the ball and is set to 1 damage per hit, buff dmg and pet dmg doesn't push and is set to 0.
That way Target Purification Graph will show who contributed how much to push.

## Giants phase

Wipes to giants are notoriously uncomfortable to deduct, it's hard to deduce damage and cc contribution

I moved giants from "trash" to "targets" and added separate "Giants" phase to easily display stats having to do with clearing giants in timely manner.

### CODE REVIEW

File: GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs [hunk 1]
```diff
@@ -496,10 +518,36 @@ internal override void EIEvtcParse(ulong gw2Build, FightData fightData, AgentDat
...
+                        {
+                            pushableOrbAgents.Add(target.AgentItem.Agent);
+                        }
                         target.OverrideName("Heart " + (++purificationID));
                         break;
                 }
             }
+
+            // Overwrite target damage in push phases to show number of strike hits by players as the main metric of the phase
+            foreach (CombatItem cbt in combatData)
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:531]: Add a PlayerSrcAllHitsMechanic instead, checking evt.To species to match orbs. You can see it used in KC in order to check hits against the core.
Goota say, what you are doing here is a little bit too extreme for my taste xD

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:531]: This also hide actual damage done against the orb in the phase where it needs to be killed.

Comment (asterius1) [reply GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs]: Kinda agree it is a hack and didn't think about displaying it as mechanic. I did a quick rebase deleted commit you commented on and just added orb push as mechanic the way you recommended it. Still readable enough on graph so good for me.

**Comment by asterius1** [comment]
redone the orb push to use PlayerSrcAllHitsMechanic

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**